cli/sawtooth_cli/network_command/peers.py

```diff
@@ -0,0 +1,152 @@
+# Copyright 2018 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+
+import os
+import json
+import shlex
+import subprocess
+
+from sawtooth_cli.exceptions import CliException
+
+
+DOT_FILE = 'peers.dot'
+PNG_FILE = 'peers.png'
+
+
+def add_peers_parser(subparsers, parent_parser):
+    parser = subparsers.add_parser(
+        'peers',
+        help='TODO',
+        description='TODO',
+    )
+
+    peers_parsers = parser.add_subparsers(
+        title='subcommands',
+        dest='peers_command')
+    peers_parsers.required = True
+
+    _add_list_parser(peers_parsers)
+    _add_graph_parser(peers_parsers)
+
+
+def _add_list_parser(parser):
+    list_parser = parser.add_parser(
+        'list',
+        help='List peers for validators with given URLs',
+        description=(
+            'List peers for validators with given URLs separated by commas.'))
+
+    list_parser.add_argument(
+        'urls',
+        nargs=1,
+        help='the comma-separated URLs of the validators to be queried',
+        type=str)
+
+    list_parser.add_argument(
+        '--pretty', '-p',
+        action='store_true',
+        help='Pretty-print the results')
+
+
+def do_peers(args):
+    if args.peers_command == 'list':
+        _do_peers_list(args)
+    elif args.peers_command == 'graph':
+        _do_peers_graph(args)
+    else:
+        raise CliException('Invalid command: {}'.format(args.subcommand))
+
+
+def _do_peers_list(args):
+    urls = args.urls[0].split(',')
+
+    print(
+        json.dumps(
+            _get_peer_endpoints(urls),
+            sort_keys=True,
+            indent=(4 if args.pretty else 0)
+        )
+    )
+
+
+def _get_status(url):
+    status = json.loads(
+        subprocess.check_output(
+            shlex.split(
+                'sawtooth status show --url {}'.format(url))
```

Comment:
This should not drop into a shell to execute this. You should do the request manually using the requests library.